### PR TITLE
Add some extra introspection functionality

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

**Comment by rhornung67** [comment]
This is very introspective of you... 😄

File: src/umpire/strategy/AllocationTracker.cpp [hunk 1]
```diff
@@ -61,7 +61,14 @@ AllocationTracker::getHighWatermark() const noexcept
 std::size_t
 AllocationTracker::getActualSize() const noexcept
 {
-  return m_allocator->getActualSize();
+  return (m_allocator->getActualSize() > 0) ? 
+    m_allocator->getActualSize() : m_current_size;
+}
```
Comment (mcfadden8) [inline src/umpire/strategy/AllocationTracker.cpp:66]: Not sure if the compiler will wind of up calling method twice or not.

```suggestion
auto actual_size = m_allocator->getActualSize();
return actual_size > 0 ? actual_size : m_current_size;
```

File: src/umpire/strategy/DynamicPoolMap.cpp [hunk 2]
```diff
@@ -416,6 +432,16 @@ std::size_t DynamicPoolMap::releaseFreeBlocks()
...
+#if defined(UMPIRE_ENABLE_BACKTRACE)
+  if (released_bytes > 0) {
+    umpire::util::backtrace bt{};
+    umpire::util::backtracer<>::get_backtrace(bt);
+    UMPIRE_LOG(Info, "actual_size: " << m_actual_bytes 
+      << " (prev: " << (m_actual_bytes+released_bytes) 
+      << ") " << umpire::util::backtracer<>::print(bt));
+  }
+#endif
+
```
Comment (mcfadden8) [inline src/umpire/strategy/DynamicPoolMap.cpp:444]: What about DynamicPoolList?

File: src/umpire/util/backtrace.hpp [hunk 3]
```diff
@@ -0,0 +1,151 @@
+//////////////////////////////////////////////////////////////////////////////
```
Comment (mcfadden8) [inline src/umpire/util/backtrace.hpp:1]: Shouldn't this be named backtrace.inl?

File: src/umpire/strategy/DynamicPoolMap.cpp (hunk 2 above)
Comment (davidbeckingsale) [reply src/umpire/strategy/DynamicPoolMap.cpp:444]: That's handled by the DynamicSizePool.hpp file.

File: src/umpire/util/backtrace.hpp (hunk 3 above)
Comment (davidbeckingsale) [reply src/umpire/util/backtrace.hpp:1]: Potentially. I'm not sure whether there is that much benefit to maintaining inl file for something so small.

Comment (mcfadden8) [reply src/umpire/util/backtrace.hpp:1]: Just trying to be consistent since we have other .inl files in Umpire.

**Pull request merged**